bot/tasks/pennychat.py

```diff
@@ -67,37 +66,57 @@ def post_organizer_edit_after_share_blocks(penny_chat_view_id):
     )
 
 
-@background
-def add_google_meet(penny_chat_id):
-    slack_client = get_slack_client()
-    penny_chat_invitation = PennyChatSlackInvitation.objects.get(id=penny_chat_id)
-    user = None
+def get_user_google_calendar_from_slack_id(slack_id):
+    user = get_or_create_social_profile_from_slack_id(slack_id).user
     try:
-        user = get_or_create_social_profile_from_slack_id(penny_chat_invitation.organizer_slack_id).user
         google_credentials = GoogleCredentials.objects.get(user=user)
     except GoogleCredentials.DoesNotExist:
         authorization_url = get_authorization_url(user)
+        slack_client = get_slack_client()
         slack_client.chat_postMessage(
-            channel=penny_chat_invitation.organizer_slack_id,
+            channel=slack_id,
             blocks=add_google_integration_blocks(authorization_url, from_penny_chat=True),
         )
         return
-    except SlackApiError:
-        return
 
     credentials = build_credentials(google_credentials)
-    calendar = GoogleCalendar(credentials=credentials)
+    return GoogleCalendar(credentials=credentials)
+
+
+@background
+def add_google_meet(penny_chat_id):
+    penny_chat_invitation = PennyChatSlackInvitation.objects.get(id=penny_chat_id)
+
+    calendar = get_user_google_calendar_from_slack_id(penny_chat_invitation.organizer_slack_id)
+
+    if calendar is None:
+        return
 
     meet = calendar.create_event(
         summary=penny_chat_invitation.title,
         description=penny_chat_invitation.description,
         start=penny_chat_invitation.date
     )
 
-    penny_chat_invitation.video_conference_link = meet['hangoutLink']
+    penny_chat_invitation.video_conference_link = meet.get('hangoutLink')
+    penny_chat_invitation.google_event_id = meet.get('id')
     penny_chat_invitation.save()
 
 
+@background
+def update_google_meet(penny_chat_id):
+    penny_chat_invitation = PennyChatSlackInvitation.objects.get(id=penny_chat_id)
+
+    calendar = get_user_google_calendar_from_slack_id(penny_chat_invitation.organizer_slack_id)
```

Comment:
if the user retracts google permissions then calendar will be None here, add an `if calendar is None` here or we'll error on line 112.  In this case I guess we just keep the calendar invite as is? (I presume it will still work)